### DISCUSSION TITLE
[Question] Does Prisma support distributed transaction?

### DISCUSSION THREAD

**Discussion by KokoTa** [prisma/prisma#20776] [Q&A]
## Question

If i have two nestjs services, A service send money, B service get money.

Have the way to rollback transaction when B fail?

I just know java have the `seata` package to solve it.

**Comment by jkomyno** [comment]
Hi @KokoTa, for this kind of questions, we recommend using the [`Discussions` tab](https://github.com/prisma/prisma/discussions) on this Github Repository.
I've already converted this question in a discussion for you :)

**Comment by ludralph** [comment]
Hi @KokoTa 👋 

Thank you for raising this question. 

Prisma does not natively support distributed transactions across multiple databases or services. This means that if you have two services (A and B) each with its own database, you cannot manage a transaction that spans both services directly using Prisma. For distributed transactions across multiple services, you would need to implement a solution outside of Prisma.

You can also take a look at this [discussion](https://github.com/prisma/prisma/discussions/19179) to see how a user implemented one.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by KokoTa** [reply]
@ludralph Thanks for your help.But my case may different with it. My service logic is Isolated，A service just send, B service just receive. I can't connect multi databases and set transaction in same service. Dose prisma will support distributed transaction? Maybe i need to get solution outside of Prisma.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
